test/unit/Cardano/Wallet/CoinSelection/RandomSpec.hs

```diff
@@ -150,3 +163,72 @@ propErrors drg (CoinSelProp utxo txOuts) = do
+-- H0 : μ1 = μ2
+-- H1 : μ1 /= μ2
+-- We calculate test statistics T (see implementation below) and need to choose significance, α.
+-- We reject the null hypothesis that the two means are equal if
+--- |T| > (t1-α/2,ν)
+-- where (t1-α/2,ν) is critical value of t-distribution
+-- for a given significance α and ν which specifies the degrees of freedom
+-- (we use 100 sample batches so it is about 100)
+-- Let's say the significance level is α = 0.002. For a two-sided test, we compute 1 - α/2,
+-- or 1 - 0.002/2 = 0.999. If the absolute value of the test statistic is greater than
+-- the critical value (0.999), then we reject the null hypothesis.
+--
+-- Critical values of Student's t distribution with ν degrees of freedom
+--     Probability less than the critical value (t1-α,ν)
+--
+--  ν         0.90    0.95   0.975    0.99   0.995   0.999
+--
+--  1.        3.078   6.314  12.706  31.821  63.657 318.313
+--  2.        1.886   2.920   4.303   6.965   9.925  22.327
+--  3.        1.638   2.353   3.182   4.541   5.841  10.215
+--
+--  97.       1.290   1.661   1.985   2.365   2.627   3.176
+--  98.       1.290   1.661   1.984   2.365   2.627   3.175
+--  99.       1.290   1.660   1.984   2.365   2.626   3.175
+-- 100.       1.290   1.660   1.984   2.364   2.626   3.174
+-- infinity   1.282   1.645   1.960   2.326   2.576   3.090
+--
+-- So, for 3-sigma we expect the calculated values to be less than 3.090.
+-- In software engineering we aim for at least 6-sigma,
+-- which means α < 0.0001. For,ν=100, t-value (two-sided) is 4.053
+-- (see, https://goodcalculators.com/student-t-value-calculator/
+-- or any good statistics book) and use this conservative value in the test
+propAverageUTxO
+    :: SystemDRG
+    -> CoinSelProp
+    -> Property
+propAverageUTxO drg (CoinSelProp utxo txOuts) = do
+    isRight selection ==> let Right s = selection in prop s
+  where
+    prop seed = do
+        resultsFromSampleOne <-
+            replicateM 99 (runExceptT $ random (CoinSelectionOptions 100) utxo txOuts)
+        resultsFromSampleTwo <-
+            replicateM 99 (runExceptT $ random (CoinSelectionOptions 100) utxo txOuts)
+        let balancesAfter res =
+                V.fromList
+                $ map
+                ((toDouble . balance . (excluding utxo)) .
+                 Set.fromList . map fst . inputs)
+                (seed : rights res)
+        let mean1 = mean $ balancesAfter resultsFromSampleOne
+        let var1 = variance $ balancesAfter resultsFromSampleOne
+        let mean2 = mean $ balancesAfter resultsFromSampleTwo
+        let var2 = variance $ balancesAfter resultsFromSampleTwo
+        when (var1 /=0 && var2 /= 0) $ do
+            let testStatistics = abs $ (mean1 - mean2) / (sqrt ((var1+var2)/100))
+            testStatistics `shouldSatisfy` (< 4.053)
+    (selection,_) = withDRG drg
+        (runExceptT $ random (CoinSelectionOptions 100) utxo txOuts)
+    toDouble :: (Integral a) =>  a -> Double
+    toDouble = fromIntegral
```

Comment:
@paweljakubas It seems that the test above is more about testing that our random generator is _correct_ rather than testing anything about our coin selection algorithm here :thinking: ... 

In particular, we run 99 times the coin selection on the same `utxo` and `txOuts`, and we know from the generator that our UTxO contains **at most** 100 elements (`n <- choose (1, 100)`). So, we are really just taking two random samples out of the same set, but with a size that is too small to cause any real discrepancy (provided our random generator is unbiased, but I am pretty sure it is since it comes from `cryptonite` :sweat_smile: ... )

Also, the property the test claims to test may not be _true_ (cf Edsko's post on the matter). As a consequence, I wonder if we couldn't re-use this idea of a statistical test in a different form and make sure we test something about the evolution of the UTxO distribution across many selections. Nevertheless, I am afraid of doing duplicate work with what was done in the coin selection simulation on the legacy wallet: https://github.com/input-output-hk/cardano-wallet-legacy/tree/develop/test/unit/InputSelection

This was really time-consuming and has proven that the underlying theory / research effort was right. I am not sure about spending the necessary time (nor about the value) of porting these bits into the new implementation.